src/main/java/com/spotify/reaper/storage/CassandraStorage.java

```diff
@@ -84,7 +90,12 @@
   private PreparedStatement deleteRepairScheduleByClusterAndKsPrepStmt;
 
   public CassandraStorage(ReaperApplicationConfiguration config, Environment environment) {
-    cassandra = config.getCassandraFactory().build(environment).register(QueryLogger.builder().build());
+    
+    CassandraFactory cassandraFactory = config.getCassandraFactory();
+    // all INSERT and DELETE statement prepared in this class are idempotent
+    cassandraFactory.setQueryOptions(java.util.Optional.of(new QueryOptions().setDefaultIdempotence(true)));
```

Comment:
My understanding of `setDefaultIdempotence` is that it's only useful in the case of retry policies and speculative executions. In the case of writes it will probably have no effect, right ?